### PR TITLE
[ENG-2185] Normalize registration state messages on overview page.

### PR DESCRIPTION
**Pull request by adlius**
- Ticket: [ENG-2185]
- Feature flag: n/a

## Purpose

Update the registration overview page so that it shows state messages for registration with different `machineState`s.

## Summary of Changes

- In `app/models/registration.ts`, removed `RegistrationState` in favor of `RegistrationReviewStates`
  - As a result, removed the `state()` computed and associated logic to get the derived state of a registration.
  - Also removed the `state()` references in other components and switch to using `machineState` of the registration.
- Updated the `registratries-states` component.
  - Make use of `machineState`.
  - Added and normalized translation strings.
- Updated `mirage/factories/registration.ts`
  - Added traits moderation-related registration states.
- Updated tests.

## Side Effects

None.

[ENG-2185]: https://openscience.atlassian.net/browse/ENG-2185

### CODE REVIEW

File: mirage/factories/registration.ts [hunk 1]
```diff
@@ -74,6 +92,7 @@ const stateAttrs = {
         },
     },
     normal: {
+        machinieState: RegistrationReviewStates.Accepted,
```
Comment (futa-ikeda) [inline mirage/factories/registration.ts:95]: little typo here

**Review by futa-ikeda — APPROVED** [review]
Changes look good to me. Just had a small suggestion regarding what Registration state is "normal", but optional really

File: lib/registries/addon/components/registries-states/component.ts [hunk 2]
```diff
@@ -83,7 +91,15 @@ export default class RegistriesStates extends Component {
 
     @computed('registration.{userHasAdminPermission,state,isRoot}')
```
Comment (fabmiz) [inline lib/registries/addon/components/registries-states/component.ts:92]:
```suggestion
    @computed('registration.{userHasAdminPermission,machineState,isRoot}', 'isModeratorMode')
```

File: mirage/factories/registration.ts [hunk 3]
```diff
@@ -74,6 +92,7 @@ const stateAttrs = {
         },
     },
     normal: {
```
[the remark this answers is not in the document]
Comment (fabmiz) [reply mirage/factories/registration.ts:94]: @adlius, did you see this ☝️? I second changing this trait name.

**Comment by adlius** [comment]
@fabmiz back to you.

File: lib/registries/addon/components/registries-states/component.ts [hunk 4]
```diff
@@ -81,9 +89,17 @@ export default class RegistriesStates extends Component {
         return registeredFromId && pathJoin(baseURL, registeredFromId);
     }
 
-    @computed('registration.{userHasAdminPermission,state,isRoot}')
+    @computed('registration.{userHasAdminPermission,state,isRoot}', 'isModeratorMode')
```
Comment (fabmiz) [inline lib/registries/addon/components/registries-states/component.ts:92]: Am I missing something? `registration.state` cp is not a thing anymore.
```suggestion
    @computed('registration.{userHasAdminPermission,machineState,isRoot}', 'isModeratorMode')
```

**Review by fabmiz — APPROVED** [review]

**Pull request merged**